### PR TITLE
feat: add kube-linter CI check

### PR DESCRIPTION
**Pull request by psturc**
https://issues.redhat.com/browse/RHTAPSRE-102

TODO:
- [ ] mark the check `kube-linter` as required (needs to be done by admin/owner of this repo)

### CODE REVIEW

File: .github/workflows/kube-linter.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      # This prepares directory where github/codeql-action/upload-sarif@v1 looks up report files by default.
+      - name: Create ../results directory for SARIF report files
+        shell: bash
+        run: mkdir -p ../results
+
+      - name: Scan yaml files with kube-linter
```
Comment (gbenhaim) [inline .github/workflows/kube-linter.yaml:20]: Some of the resources we are deploying are located on remote repositories, or generated after applying a pathc, so we only get them when running `kustomize build`. I think that we should first build all the manifests, dump them to a directory and then run the check.

**Review by gbenhaim — CHANGES_REQUESTED** [review]

**Comment by gbenhaim** [comment]
thanks for working on this @psturc

**Comment by psturc** [comment]
> is there a nicer way for seeing the results https://github.com/redhat-appstudio/infra-deployments/actions/runs/5741308682/job/15561105886?pr=2197 ?

yep check the details of this check and then click on `View all branch alerts`
Screenshot 2023-08-03 at 10 21 20

Now that check is green, but it should be red if there's a new alert created for the code changed in a PR.

File: .github/workflows/kube-linter.yaml (hunk 1 above)
Comment (psturc) [reply .github/workflows/kube-linter.yaml:29]: updated [in this commit](https://github.com/redhat-appstudio/infra-deployments/pull/2197/commits/36cbf1e1b367b84bb94c5ad9e1da94a2e0a82cc6), please check (now we have [470 warnings](https://github.com/redhat-appstudio/infra-deployments/security/code-scanning?query=pr%3A2197+tool%3Akube-linter+is%3Aopen), yay :) 

however there were numerous errors hit [during `kustomize build` step](https://github.com/redhat-appstudio/infra-deployments/actions/runs/5750076950/job/15586110735?pr=2197#step:5:7) (search for `^ ERROR`) - should we address them in this PR or create a follow-up ticket?

Comment (gbenhaim) [reply .github/workflows/kube-linter.yaml:29]: I suggest to address them in a follow up pr.

**Review by gbenhaim — APPROVED** [review]

**Comment by gbenhaim** [comment]
well done @psturc thank you for working on this.

**Comment by gbenhaim** [comment]
@psturc tide won't let me merge it as long as the new status check fails. Can you please change if for now to always succeed (just until all the teams will fix their erros).

**Comment by gbenhaim** [comment]
@psturc is there a way to download the sarif test report?

**Comment by psturc** [comment]
> @psturc is there a way to download the sarif test report?
I didn't find a way to download the sarif test report, but if I understand it correctly, the report contains the same data as what's on the [job output](https://github.com/redhat-appstudio/infra-deployments/actions/runs/5781240056/job/15666067853?pr=2197#step:6:50)

> @psturc tide won't let me merge it as long as the new status check fails. Can you please change if for now to always succeed (just until all the teams will fix their erros).
[done](https://github.com/redhat-appstudio/infra-deployments/pull/2197/commits/d470f0cf62f0f6bdbbb51aee08fe83c344411e61)

File: .github/workflows/kube-linter.yaml (hunk 1 above)
Comment (psturc) [reply .github/workflows/kube-linter.yaml:29]: https://issues.redhat.com/browse/RHTAPSRE-109

**Pull request merged**